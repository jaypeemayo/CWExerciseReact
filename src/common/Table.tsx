import * as React from "react";

export interface ITable {
    children: any;
    columnNames: string [];
    sortedColumnName: string;
    isAscending: boolean
    onClickHeader(columName: string): void;
}
export const Table = (props: ITable) => {
    return <table className="table thead-light table-hover">
        <thead className="thead-light">
            <tr>
                {
                    props.columnNames.map((columName, i) => <th key={i} onClick={()=>{props.onClickHeader(columName);}}>{columName}</th>)
                }
            </tr>
        </thead>
        <tbody>
            {props.children}
        </tbody>
    </table>

}
