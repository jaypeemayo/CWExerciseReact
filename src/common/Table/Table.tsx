import * as React from "react";
import { FaChevronUp, FaChevronDown } from "react-icons/fa";
import 'bootstrap/dist/css/bootstrap.min.css';
import BTable from 'react-bootstrap/Table';
import { ITable } from ".";


const renderChevron = (props: ITable, columName: string) => {
    if (props.sortedColumnName === columName) {
        if (props.isAscending) {
            return <FaChevronDown></FaChevronDown>;
        } else {
            return <FaChevronUp></FaChevronUp>;
        }
    }
};
export const Table = (props: ITable) => {
    return (<BTable striped bordered hover>
        <thead className="thead-light">
            <tr>
                {
                    props.columnNames.map((columName, i) => (<th key={i} onClick={() => {
                        props.onClickHeader(columName);
                    }}>
                        <a href="javascript:void(0)" >
                            <span className="px-1">{columName}</span>
                            {renderChevron(props, columName)}
                        </a>

                    </th>))
                }
            </tr>
        </thead>
        <tbody>{props.children}</tbody>
    </BTable>);
};
