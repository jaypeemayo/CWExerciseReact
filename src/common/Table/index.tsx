export interface ITable {
    children: any;
    columnNames: string[];
    sortedColumnName: string;
    isAscending: boolean;
    onClickHeader(columName: string): void;
}
