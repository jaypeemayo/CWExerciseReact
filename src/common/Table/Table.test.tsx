import { render } from "@testing-library/react";
import * as React from "react";
import { Table } from "./Table";



jest.mock("react-icons/fa", () => {
    return {
        FaChevronUp: (props: any) => {
            return <div>Mock Chevron Up</div>;
        },
        FaChevronDown: (props: any) => {
            return <div>Mock Chevron Down</div>;
        },
    };
});

test("Test snapshot", () => {
    const onClickHeader = jest.fn();
    const {container, unmount} = render(
        <Table  columnNames={["Column A", "Column B"]} sortedColumnName={"Column A"} isAscending = {true} onClickHeader={onClickHeader}>
           {test}
        </Table>
        ,
    );

    expect(container).toMatchSnapshot();
    unmount();
});


