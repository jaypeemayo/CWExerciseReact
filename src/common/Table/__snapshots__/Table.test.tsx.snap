// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<div>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead
      class="thead-light"
    >
      <tr>
        <th>
          <a
            href="javascript:void(0)"
          >
            <span
              class="px-1"
            >
              Column A
            </span>
            <div>
              Mock Chevron Down
            </div>
          </a>
        </th>
        <th>
          <a
            href="javascript:void(0)"
          >
            <span
              class="px-1"
            >
              Column B
            </span>
          </a>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
